os/fs/smartfs/smartfs.h

```diff
@@ -311,6 +313,7 @@ struct smartfs_ofile_s {
 #ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
 	uint8_t *buffer;			/* Sector buffer to reduce writes */
 	uint8_t bflags;				/* Buffer flags */
+	uint16_t last_psector;			/* Last sector in old parent directory chain, will be used to update chain after writing entry */
```

Comment:
it is useful for only entry, I think you'd better expand entry.
```
struct smartfs_entry_s {
	uint16_t firstsector;		/* Sector number of the name */
	uint16_t dsector;			/* Sector number of the directory entry */
	uint16_t doffset;			/* Offset of the directory entry */
	uint16_t dfirst;			/* 1st sector number of the directory entry */
        uint16_t dprev;                /* Previous entry sector that is need to be chained with our parent */
	uint16_t flags;				/* Flags, including mode */
	FAR char *name;				/* inode name */
	uint32_t utc;				/* Time stamp */
	uint32_t datlen;			/* Length of inode data */
};
```
or keep current and below(we optimized read(memcpy in progmem) performance of target board 3times faster)
```
if (sf->entry->dsector != sf->entry->dfirst) {
     smartfs_setbuffer(&readwrite, entry->dsector, 0, fs->fs_llformat.availbytes, (uint8_t *)fs->fs_rwbuffer);
     FS_IOCTL(READ);
     while (nextsector_is_not_0xff) {
        if (SMARTFS_NEXTSECTOR(header) == 0xff) {
            write_next_header;
        }
      }
}
```

Reply:
Adding last_psector to smartfs_entry_s seems meaningless.
last_psector has meaning only till the new sector is not chained to the parent directory.

Regarding the loop that you suggested, I made some more changes to align code with writeentry.
So last_psector is needed to store parendirsector because it is passed as a parameter to writeentry.
Could you please take a look once again?

Reply:
Add to entry is correct, in my opinion, because  parent sector(previous) sector's is necessary only for part of entry process.

Reply:
please check above comment to.

Reply:
Changing this now